tensorflow/c/experimental/gradients/BUILD

```diff
@@ -155,6 +155,73 @@ filegroup(
     ],
 )
 
+cc_library(
+    name = "gradient_checker",
+    testonly = 1,
+    srcs = [
+        "gradient_checker.cc",
+    ],
+    hdrs = [
+        "gradient_checker.h",
+    ],
+    visibility = [
+        "//tensorflow:internal",
+    ],
+    deps = [
+        ":unified_api_testutil",
+        "//tensorflow/c/eager:abstract_tensor_handle",
+        "@com_google_absl//absl/types:span",
+        "//tensorflow/c/experimental/ops:math_ops",
+        "//tensorflow/core/platform:tensor_float_32_utils",
+    ] + if_libtpu(
+        if_false = ["//tensorflow/compiler/mlir/tensorflow/c:mlir_c_api_registration"],
+        if_true = [],
+    ),
+)
+
+tf_cuda_cc_test(
+    name = "gradient_checker_test",
+    size = "small",
+    srcs = [
+        "gradient_checker_test.cc",
+    ],
+    args = ["--heap_check=local"],
+    linkstatic = tf_kernel_tests_linkstatic(),
+    tags = tf_cuda_tests_tags() + [
+        "nomac",
+        "manual",  #TODO(vnvo2409): Remove
```

Comment:
I doubt we will be able to submit code that does not compile, even with this tag, so we should probably address this or find an alternative way of refactoring that does not require checking in broken code.

Reply:
we could refactor it in this PR but it makes this PR become much larger. How about deleting this file/rule and add it again in the next PR ?

Reply:
Are the changes non-trivial? I believe the main change would be remove the `registry` arg from the model functions and use the new `RunModel` function. To make this easier to review we could leave the file locations the same for now and just focus on the symbols. What do you think?

Reply:
The changes are trivial ( It's really just copy-paste ). I would like to add some more tests to `gradient_checker_test` but let's just leave it here now. About the moving file locations, I think Github is smart enough to not show the diff when renaming ( apparently, Github decide that `gradient_checker_test` is deleted and created in somewhere else rather than renamed. I think it is because I removed a lof of stuff out of it ). 

Please tell me if you have difficulties in reviewing this PR, I will try to make it easier.